cmd/bootkube/recover.go

```diff
@@ -0,0 +1,108 @@
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/kubernetes-incubator/bootkube/pkg/recovery"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cmdRecover = &cobra.Command{
+		Use:          "recover",
+		Short:        "Recover a control plane from state stored in etcd.",
+		Long:         "",
+		PreRunE:      validateRecoverOpts,
+		RunE:         runCmdRecover,
+		SilenceUsage: true,
+	}
+
+	recoverOpts struct {
+		assetDir            string
+		etcdCAPath          string
+		etcdCertificatePath string
+		etcdPrivateKeyPath  string
+		etcdServers         string
+		etcdPrefix          string
+		kubeConfigPath      string
+	}
+)
+
+func init() {
+	cmdRoot.AddCommand(cmdRecover)
+	cmdRecover.Flags().StringVar(&recoverOpts.assetDir, "asset-dir", "", "Output path for writing recovered cluster assets.")
+	cmdRecover.Flags().StringVar(&recoverOpts.etcdCAPath, "etcd-ca-path", "", "Path to an existing PEM encoded CA that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-certificate-path and --etcd-private-key-path, and must have etcd configured to use TLS with matching secrets.")
+	cmdRecover.Flags().StringVar(&recoverOpts.etcdCertificatePath, "etcd-certificate-path", "", "Path to an existing certificate that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-ca-path and --etcd-private-key-path, and must have etcd configured to use TLS with matching secrets.")
+	cmdRecover.Flags().StringVar(&recoverOpts.etcdPrivateKeyPath, "etcd-private-key-path", "", "Path to an existing private key that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-ca-path and --etcd-certificate-path, and must have etcd configured to use TLS with matching secrets.")
+	cmdRecover.Flags().StringVar(&recoverOpts.etcdServers, "etcd-servers", "", "List of etcd servers URLs including host:port, comma separated. Cannot be specified with --etcd-data-dir.")
+	cmdRecover.Flags().StringVar(&recoverOpts.etcdPrefix, "etcd-prefix", "/registry", "Path prefix to Kubernetes cluster data in etcd.")
+	cmdRecover.Flags().StringVar(&recoverOpts.kubeConfigPath, "kubeconfig", "", "Path to kubeconfig for communicating with the cluster.")
+}
+
+func runCmdRecover(cmd *cobra.Command, args []string) error {
+	var err error
+	recoverOpts.kubeConfigPath, err = filepath.Abs(recoverOpts.kubeConfigPath)
+	if err != nil {
+		return err
+	}
+	etcdClient, err := createEtcdClient()
+	if err != nil {
+		return err
+	}
+	as, err := recovery.Recover(context.Background(), recovery.NewEtcdBackend(etcdClient, recoverOpts.etcdPrefix), recoverOpts.kubeConfigPath)
+	if err != nil {
+		return err
+	}
+	return as.WriteFiles(recoverOpts.assetDir)
+}
+
+func validateRecoverOpts(cmd *cobra.Command, args []string) error {
+	if recoverOpts.assetDir == "" {
+		return errors.New("Missing required flag: --asset-dir")
```

Comment:
Done for all (I was following the other command definitions, but personally I prefer the go convention)